libcxx/include/__type_traits/has_unique_object_representation.h

```diff
@@ -22,7 +23,12 @@ _LIBCPP_BEGIN_NAMESPACE_STD
 
 template <class _Tp>
 struct _LIBCPP_TEMPLATE_VIS has_unique_object_representations
-    : public integral_constant<bool, __has_unique_object_representations(_Tp)> {};
+    // TODO: We work around a Clang bug in __has_unique_object_representations by instantiating the
+    //       builtin on the non-array type first and discarding that. This is issue #95311.
+    //       This workaround can be removed once the bug has been fixed in all supported Clangs.
+    : public integral_constant<bool,
+                               ((void)__has_unique_object_representations(remove_all_extents_t<_Tp>),
+                                __has_unique_object_representations(_Tp))> {};
```

Comment:
Can't we just do `__has_unique_object_representations(remove_all_extents_t<_Tp>)` instead of the `operator,` weirdness?

Reply:
Note that GCC 14 crashes without this workaround: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=115476 so this will have to remain until we drop support for Clang 18 and GCC 14 (if this gets fixed in Clang 19 and GCC 15)